src/tx-builder/services/MerkleDistributor.ts

```diff
@@ -152,6 +154,40 @@ export default class MerkleDistributor extends BaseService<MerkleDistributorV1>
     return this.contract.hasPendingRoot();
   }
 
+  public async getUserBalancesPerEpoch(): Promise<UserRewardsBalancesPerEpoch> {
+    const activeRootData: ActiveRootDataAndHistory = await this.getActiveRootData();
+    return activeRootData.userBalancesPerEpoch;
+  }
+
+  public async getActiveUsersInEpoch(epoch: number): Promise<string[]> {
+    const userBalancesPerEpoch: UserRewardsBalancesPerEpoch = await this.getUserBalancesPerEpoch();
+
+    // if epoch has not occurred then return
+    // NOTE: first epoch is zero
+    if (Object.keys(userBalancesPerEpoch).length <= epoch) {
+      return [];
+    }
+
+    // in epoch zero any user with a balance had to have been active
+    if (epoch === EPOCH_ZERO) {
+      return Object.keys(userBalancesPerEpoch[EPOCH_ZERO]);
+    }
+
+    return _.chain(Object.keys(userBalancesPerEpoch[epoch]))
+      .filter((address: string) => {
```

Comment:
Maybe add a comment that we assume the addresses are stored as checksummed in IPFS (since we read all of this address data from IPFS)